kategory/src/main/kotlin/kategory/instances/Function1Instances.kt

```diff
@@ -0,0 +1,31 @@
+package kategory
+
+interface Function1Instances<P> :
+        Functor<Function1.F>,
+        Applicative<Function1.F>,
+        Monad<Function1.F>,
+        MonadReader<Function1.F, P> {
+
+    override fun ask(): HK<Function1.F, P> =
```

Comment:
The compiler doesn't have enough information about the generic F in this context, so it fails.